ion-schema/src/system.rs

```diff
@@ -763,7 +763,7 @@ impl Resolver {
 
         // add all types from pending_types to type_store
         pending_types.update_type_store(type_store, None, &isl_type_names)?;
-        Ok(Schema::new(id, Rc::new(type_store.to_owned())))
+        Ok(Schema::new(id, Arc::new(type_store.to_owned())))
```

Comment:
Is `type_store.to_owned()` equivalent to `type_store.clone()` here?

If so, it looks like this is a deep copy. Should this function have ownership of `type_store` instead of holding a `&mut TypeStore`? Or maybe we should be cloning an `Arc` instead of trying to own the type store?

Reply:
See https://github.com/amazon-ion/ion-schema-rust/pull/153#discussion_r1154887253

Reply:
This case is actually slightly different than the other one. Earlier on in this function, the typestore is created like this:
```rust
let type_store = &mut TypeStore::default();
```
But instead, we should be able to do this, which would avoid the need to clone into the `Arc`.
```rust
let mut type_store = TypeStore::default();
```